### PR TITLE
Add tests for async_handler

### PR DESCRIPTION
**Pull request by kopylovanat**
Tests for [2.6. sycl::async_handler](https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/test_plans/exceptions.asciidoc#26-syclasync_handler)

### CODE REVIEW

File: tests/exceptions/exceptions_async_handler.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  }
+};
+
+TEST_CASE("Check that sycl::async_handler is expected type", "[exception]") {
+  CHECK(std::is_same_v<sycl::async_handler,
+                       std::function<void(sycl::exception_list)>>);
+}
+
+TEST_CASE(
+    "Check that when there are no exceptions expected then the async handler "
```
Comment (keryell) [inline tests/exceptions/exceptions_async_handler.cpp:31]:
```suggestion
    "Check that, when there is no exception expected, the async handler "
```

File: tests/exceptions/exceptions_async_handler.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  SECTION("event::wait_and_throw()") {
+    auto event = q.submit([&](sycl::handler &h) {});
+    event.wait_and_throw();
+    CHECK_FALSE(asyncHandler.invoked);
+  }
+
+  SECTION("static event::wait_and_throw(const std::vector<event> &eventList)") {
+    auto event = q.submit([&](sycl::handler &h) {});
+    std::vector<sycl::event> event_vector(1, event);
```
Comment (keryell) [inline tests/exceptions/exceptions_async_handler.cpp:57]: Think Python :-)
```suggestion
    std::vector event_vector { event };
```

**Review by keryell — CHANGES_REQUESTED** [review]
Looks good.

**Review by ProGTX — APPROVED** [review]

File: tests/exceptions/exceptions_async_handler.cpp (hunk 1 above)
Comment (kopylovanat) [reply tests/exceptions/exceptions_async_handler.cpp]: Done.

File: tests/exceptions/exceptions_async_handler.cpp (hunk 2 above)
Comment (kopylovanat) [reply tests/exceptions/exceptions_async_handler.cpp]: Thank you! Done.

**Review by keryell — APPROVED** [review]
Thanks.

**Pull request merged**